src/main/java/bio/terra/service/dataset/DatasetJsonConversion.java

```diff
@@ -138,6 +145,124 @@ public static DatasetModel populateDatasetModelFromDataset(
           metadataDataAccessUtils.accessInfoFromDataset(dataset, userRequest));
     }
 
+    if (include.contains(DatasetRequestAccessIncludeModel.SNAPSHOT_BUILDER_CONFIG)) {
+      datasetModel.snapshotBuilderConfig(
```

Comment:
So this is interesting. This util class has no state but it's supposed to populate the dataset response with information that isn't in the dataset object. It currently works around this by having the caller pass in a service object `metadataDataAccessUtils` which it uses to get that information. This is not a great pattern, since you really shouldn't pass a service object around this way.

The way I look at this, you have two main options. Either we inject the service into the caller (`DatasetService`), and (conditionally) get the data there, or we get the data in this method (the callee).

To get the data here, you can either continue along this path; you could inject the new service into the caller and  pass it to this method as another argument. Or, you could change `DatasetJsonConversion` to be a `@Component` and it can inject the services it needs directly.

I guess a third option is that the information would become a first class member of `Dataset` itself and would be retrieved by the `DatasetDao.retrieve()` code.

IMO I would convert this class to a `@Component` and have it inject its state directly rather than having `DatasetService` pass it in. I think this change is pretty straightforward but may require a lot of test changes.

Reply:
> IMO I would convert this class to a @Component and have it inject its state directly rather than having DatasetService pass it in. I think this change is pretty straightforward but may require a lot of test changes.

Actually this wasn't hard to do, let me know if you're interested and I can give you the diffs.

Reply:
I think I can try to do it myself :D